cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -1010,11 +1012,32 @@ public CloudServiceInstance getServiceInstance(String serviceName, boolean requi
                             .collect(Collectors.toList());
     }
 
-    private Map<String, Metadata> getServiceInstancesMetadata(List<String> names) {
+    private Map<String, Metadata> getServiceInstancesMetadata(List<String> serviceInstanceNames) {
+        Map<String, Metadata> serviceInstancesMetadata = new HashMap<>();
+        for (List<String> batchOfServiceInstanceNames : toBatches(serviceInstanceNames, BATCH_SIZE_FOR_FILTER_IN_METADATA_REQUESTS)) {
+            serviceInstancesMetadata.putAll(getServiceInstancesMetadataWithoutBatching(batchOfServiceInstanceNames));
+        }
+        return serviceInstancesMetadata;
+    }
+
+    private <T> List<List<T>> toBatches(List<T> largeList, int batchSize) {
+        List<List<T>> batches = new ArrayList<>();
+        AtomicInteger counter = new AtomicInteger();
+        for (T t : largeList) {
```

Comment:
Why not use a regular `for` loop and just add a new list whenever `i % batchSize == 0`?